custodia/httpd/server.py

```diff
@@ -161,9 +161,15 @@ def peer_creds(self):
         if self.request.family != socket.AF_UNIX:
             self._creds = None
             return self._creds
+        # pid_t: signed int32, uid_t/gid_t: unsigned int32
+        fmt = 'iII'
         creds = self.request.getsockopt(socket.SOL_SOCKET, SO_PEERCRED,
-                                        struct.calcsize('3i'))
-        pid, uid, gid = struct.unpack('3i', creds)
+                                        struct.calcsize(fmt))
+        pid, uid, gid = struct.unpack(fmt, creds)
+        if uid == (1<<32)-1:
+            uid = -1
```

Comment:
Why this conversion ?
I do not see any special meaning assigned to -1 through the code.